services/graphql-server/src/graphql/resolvers/platform/content.js

```diff
@@ -659,7 +662,27 @@ module.exports = {
         maxResults,
         ...(pageToken && { pageToken }),
       };
-      return googleDataApiClient.request('youtube.playlistItems', payload);
+      const response = await googleDataApiClient.request('youtube.playlistItems', payload);
+      const find400 = [];
+      if (getAsArray(response, 'errors')) { // If the response contains errors
+        const errors = getAsArray(response, 'errors');
+        errors.forEach((error) => {
+          if (error.extensions.exception.statusCode >= 400) { // If there is a 400 error
+            find400.push(error);
+          }
+        });
+      }
+      if (find400.length > 0) {
+        return { pageInfo: {}, items: [] }; // If a 400 error is found return "nothing"
+      }
+      const initialVideoList = getAsArray(response, 'items');
+      const filteredVideoList = [];
+      initialVideoList.forEach((video) => {
+        if (video.status.privacyStatus === 'public') { // Check to make sure each video is public status
+          filteredVideoList.push(video);
+        }
+      });
+      return { pageInfo: getAsObject(response, 'pageInfo'), items: filteredVideoList };
```

Comment:
No need to define additional arrays here, just filter the response. Ideally this should be a input option to include private videos (defaulting to `false`).

```suggestion
      return {
        pageInfo: getAsObject(response, 'pageInfo'),
        items: getAsArray(response, 'items').filter((video) => showPrivate || get(video, 'status.privacyStatus') === 'public'),
      };
```